### PR TITLE
hero buffs

### PR DESCRIPTION
**Pull request by POCKETMONEY99**
Buffs to Chen, Clinkz, crystal madien, dark willow, death prohpet, disruptor, Doom, ember spirit, enigma, legion commander, lina, lcyan, mirana, night stalker, Razor, Sand king, Tusk and Treant Protector

### CODE REVIEW

File: game/scripts/npc/abilities/winter_wyvern_splinter_blast.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
     // Damage.
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityDamage"                                       "100 180 260 340 740 1140"
+    "AbilityDamage"                                       "140 220 300 380 800 1200"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/winter_wyvern_splinter_blast.txt]: this needs // OAA

File: game/scripts/npc/abilities/chen_hand_of_god.txt [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
     // Time
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityCooldown"                                     "80" //OAA
+    "AbilityCooldown"                                     "80 70 60 60 60" //OAA
```
[the remark this answers is not in the document]
Comment (POCKETMONEY99) [reply game/scripts/npc/abilities/chen_hand_of_god.txt]: The heal is alot yeah but its compensatiting for the fact that his E is pretty useless in OAA and the rest of his basic spells arent great. Also considering that your 6 slotted at 15 minutes max 450/600 heal isnt great. Ill leave the cd how it was.  

additionally im comparing this spell to other damage mitiation ults like aghs song of the siren, guardian angel and ghostship and it feels fairly underwhelming.

File: game/scripts/npc/abilities/death_prophet_silence.txt [hunk 3]
```diff
@@ -23,11 +23,11 @@
     // Time
     //-------------------------------------------------------------------------------------------------------------
     "AbilityCooldown"                                     "15 14 13 12 12 12"
-    "AbilityDuration"                                     "3.0 3.5 4.0 4.5 5.0 5.5" //OAA
+    "AbilityDuration"                                     "3.0 4.0 5.0 6.0 6.0 6.0"
```
[the remark this answers is not in the document]
Comment (POCKETMONEY99) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: But it is 6 seconds in normal dota. I could see why they nerfed it from 6/8/11 at OAA levels but making an already average spell into an even worse spell doesnt make sense to me.

[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: I just compared it with Drow Ranger gust. Same scaling.

File: game/scripts/npc/abilities/death_prophet_silence.txt [hunk 4]
```diff
@@ -23,11 +23,11 @@
...
 
     // Cost
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityManaCost"                                     "80 90 100 110 165 275"
+    "AbilityManaCost"                                     "80 75 70 65 60 55" //OAA
```
[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: I changed my mind. This is fine.

File: game/scripts/npc/abilities/death_prophet_silence.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: With max level Octarine Core, DP can perma-silence people.  Also if its the same duration on lvl 5 and 6, there would be no reason to upgrade ability to level 5 or 6. Please revert. Instead of buffing duration: You can make the cd 12 seconds on every level.

**Comment by DarkoniusXNG** [comment]
Major backlash xD

**Comment by lovesmuffins** [comment]
Clinkz Army CD's could probably be further reduced to a 90/80/70/60/50 as I don't think it has much impact outside of duel and I feel like the further cd buff would allow it to be used for bosses.

File: game/scripts/npc/abilities/death_prophet_silence.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (lovesmuffins) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: I'd still suggest a revert just because this isn't a buff and seems like a random number change as DP does not and should not have mana issues whatsoever. I think the cd buff is good enough to make this ability better and would just suggest making the manacost what it is currently,

File: game/scripts/npc/abilities/death_prophet_silence.txt [hunk 5]
```diff
@@ -42,7 +42,6 @@
       {
         "var_type"                                        "FIELD_FLOAT"
         "duration"                                        "3.0 3.5 4.0 4.5 5.0 5.5" //OAA
-      }
```
Comment (lovesmuffins) [inline game/scripts/npc/abilities/death_prophet_silence.txt]: why did you remove this line?

File: game/scripts/npc/abilities/doom_bringer_devour.txt [hunk 6]
```diff
@@ -45,7 +45,7 @@
       "03"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "regen"                                           "10 20 30 40 70 100" //OAA
+        "regen"                                           "8 16 24 32 70 100" //OAA
```
Comment (lovesmuffins) [inline game/scripts/npc/abilities/doom_bringer_devour.txt]: I would probably nerf the last 2 levels as well. Something like 50/80 instead of 70/100

File: game/scripts/npc/abilities/death_prophet_silence.txt (hunk 5 above)
Comment (lovesmuffins) [inline game/scripts/npc/abilities/death_prophet_silence.txt]: why did you remove this line?

File: game/scripts/npc/abilities/death_prophet_silence.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (lovesmuffins) [reply game/scripts/npc/abilities/death_prophet_silence.txt]: I'd still suggest a revert just because this isn't a buff and seems like a random number change as DP does not and should not have mana issues whatsoever. I think the cd buff is good enough to make this ability better and would just suggest making the manacost what it is currently,

File: game/scripts/npc/abilities/clinkz_burning_army.txt [hunk 7]
```diff
@@ -22,7 +22,7 @@
 
     // Time
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityCooldown"                                     "110"
+    "AbilityCooldown"                                     "90" //OAA
```
Comment (lovesmuffins) [inline game/scripts/npc/abilities/clinkz_burning_army.txt]: Clinkz Army CD's could probably be further reduced to a 90/80/70/60/50 as I don't think it has much impact outside of duel and I feel like the further cd buff would allow it to be used for bosses.

File: game/scripts/npc/abilities/legion_commander_duel.txt [hunk 8]
```diff
@@ -55,7 +55,7 @@
       "03"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "reward_damage"                                   "10 18 26 42 58"
+        "reward_damage"                                   "20 30 40 50 60" //OAA
```
Comment (lovesmuffins) [inline game/scripts/npc/abilities/legion_commander_duel.txt]: don't double level 1 reward

You're underestimating how strong legion can be with these buffs.

I also suggest we cap the duration to 6 because 7 second duel is fucking stupid to be fair.
Duration: 4.5/5/5.5/6/6
Reward damage: 12/24/36/48/60

Honestly you might just want to leave the duration as it is and just chang reward damage to what I put.

File: game/scripts/npc/abilities/lycan_feral_impulse.txt [hunk 9]
```diff
@@ -22,7 +22,7 @@
       "02"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "bonus_damage"                                    "12 24 36 48 60 85"
+        "bonus_damage"                                    "15 30 45 60 75 100" //OAA
```
[the remark this answers is not in the document]
Comment (lovesmuffins) [reply game/scripts/npc/abilities/lycan_feral_impulse.txt]: Lycan is pretty scary and would be played if his wolves weren't bugged out (Like how silencer would be a top hero if his W wasn't buffed). I suggest keeping it at base values as it is until we fix wolves and see where's he at from there.

File: game/scripts/npc/abilities/lycan_shapeshift.txt [hunk 10]
```diff
@@ -16,7 +16,7 @@
     "MaxLevel"                                            "5"
     // Time
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityCooldown"                                     "130 105 80 80 80"
+    "AbilityCooldown"                                     "80 70 60 60 60" //OAA
```
[the remark this answers is not in the document]
Comment (lovesmuffins) [reply game/scripts/npc/abilities/lycan_shapeshift.txt]: With the bonus duration talent lowering the cd seems pretty dumb. I'd suggest just buffing his level durations to like 18/21/24/27/30 and maybe just buffing level 1 cd to like 110 level 2 to 100 and keeping the last 3 levels at 80 so like 110/100/80/80/80

File: game/scripts/npc/abilities/sandking_epicenter.txt [hunk 11]
```diff
@@ -52,7 +52,7 @@
       "03"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "epicenter_damage"                                "110 110 110 220 440"
+        "epicenter_damage"                                "110 130 150 300 500" //OAA
```
Comment (lovesmuffins) [inline game/scripts/npc/abilities/sandking_epicenter.txt]: I'd probably also nerf some of other Sand King's spells (like sandstorm aoe maybe) to combat these pretty big buffs to his ulti.

File: game/scripts/npc/abilities/legion_commander_duel.txt (hunk 8 above)
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/legion_commander_duel.txt]: He is increasing duration because of stupid status resistance.

File: game/scripts/npc/abilities/sandking_epicenter.txt (hunk 11 above)
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/sandking_epicenter.txt]: Sand King's ult is his worst spell. That's why it needs a buff.

Comment (lovesmuffins) [reply game/scripts/npc/abilities/sandking_epicenter.txt]: yes and literally everything else about sand king is overpowerd. a 2k range stun, a 500 huge aoe damaging spell + invis and the ability to explode early creep camps with phase boots. Nerfs to those spells are needed if we want to buff SK ult's

**Comment by lovesmuffins** [comment]
/home/travis/build/OpenAngelArena/oaa/game/scripts/npc/abilities/items/necronomicon_warrior_last_will.txt
  ✖ explosion should inherit basic dota values (600 700 800 vs 550 675 800)
    operator: equal
    expected: '550 675 800'

File: game/scripts/npc/abilities/legion_commander_duel.txt (hunk 8 above)
Comment (lovesmuffins) [reply game/scripts/npc/abilities/legion_commander_duel.txt]: you can just code it to not be affected by status resistance.

**Comment by POCKETMONEY99** [comment]
> 
> 
> /home/travis/build/OpenAngelArena/oaa/game/scripts/npc/abilities/items/necronomicon_warrior_last_will.txt
> ✖ explosion should inherit basic dota values (600 700 800 vs 550 675 800)
> operator: equal
> expected: '550 675 800'

I have no idea why this has shown up. I haven't touched nerconomicon

File: game/scripts/npc/abilities/sandking_epicenter.txt (hunk 11 above)
Comment (POCKETMONEY99) [reply game/scripts/npc/abilities/sandking_epicenter.txt]: Yes he has one of the best stuns in the game. However his damage output drops off heavily. Castic finale is fine how it is, that ability is designed to allow SK to farm fast. Sandstorm is good but its not OP.

**Comment by lovesmuffins** [comment]
> > /home/travis/build/OpenAngelArena/oaa/game/scripts/npc/abilities/items/necronomicon_warrior_last_will.txt
> > ✖ explosion should inherit basic dota values (600 700 800 vs 550 675 800)
> > operator: equal
> > expected: '550 675 800'
> 
> I have no idea why this has shown up. I haven't touched nerconomicon

either way you should either change it back or put // OAA after this so Trav2s doesn't get upset.

File: game/scripts/npc/abilities/doom_bringer_devour.txt (hunk 6 above)
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/doom_bringer_devour.txt]: OAA levels are fine.

File: game/scripts/npc/abilities/lycan_shapeshift.txt (hunk 10 above)
[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/lycan_shapeshift.txt]: First level should be 100 seconds. Last level should be 60 seconds. I proposed ''100 90 80 70 60'' scaling because that doesn't mess up with power spike timings and its still a buff.

File: game/scripts/npc/abilities/nyx_assassin_vendetta.txt [hunk 12]
```diff
@@ -17,7 +17,7 @@
     "MaxLevel"                                            "5"
     // Stats
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityCooldown"                                     "70.0 60.0 50.0 50.0 50.0"
+    "AbilityCooldown"                                     "50.0 40.0 30.0 30.0 30.0" //OAA
```
[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/nyx_assassin_vendetta.txt]: Look at duration scaling: ''40 50 60 70 80''. So, with your changes: Nyx becomes Riki 2.0 at level 12 (no downtime) and at level 18 Nyx can use Vendetta multiple times during duels. Add Octarine Core cd reduction to all of this and he becomes cancer.

**Comment by DarkoniusXNG** [comment]
Something is weird with Travis. After you changed necronomicon last will dmg, travis keeps failing at some tests. Really weird. Maybe random empty spaces are causing this.

**Comment by POCKETMONEY99** [comment]
> 
> 
> Something is weird with Travis. After you changed necronomicon last will dmg, travis keeps failing at some tests. Really weird. Maybe random empty spaces are causing this.

Im not sure what happened. Because I never edited the necro units, so I just changed it back to normal dota values. Then after that the travis CI build work complete.

**Comment by lovesmuffins** [comment]
just revert changes.

**Comment by POCKETMONEY99** [comment]
> 
> 
> just revert changes.

I tried it and the build doesnt work. I might have to remake the commit. This is a last option though cuz we lose all the dicussion we had. Also people still disagree with my Lina and Lcyan buffs.

**Comment by lovesmuffins** [comment]
> > just revert changes.
> 
> I tried it and the build doesnt work. I might have to remake the pull request. This is a last option though cuz we lose all the dicussion we had. Also people still disagree with my Lina and Lcyan buffs.

travis is broken, leave as it is until chris can fixit

File: game/scripts/npc/abilities/disruptor_static_storm.txt [hunk 13]
```diff
@@ -26,7 +26,7 @@
 
     // Time
     //-------------------------------------------------------------------------------------------------------------
-    "AbilityCooldown"                                     "90 80 70 70 70"
+    "AbilityCooldown"                                     "70 65 60 60 60" //OAA
```
[the remark this answers is not in the document]
Comment (DarkoniusXNG) [reply game/scripts/npc/abilities/disruptor_static_storm.txt]: Here is an idea: 70s cd on every level.
